apps/stops/lib/route_stop.ex

```diff
@@ -99,6 +99,30 @@ defmodule Stops.RouteStop do
     do_list_from_shapes(shape.name, Enum.map(stops, & &1.id), stops, route)
   end
 
+  def list_from_shapes(shapes, stops, %Route{id: "CR-Fairmount"} = route, 0) do
+    mainline = Enum.at(shapes, 0)
+    foxboro_extension = Enum.at(shapes, 1)
+
+    distinct_stop_ids =
+      mainline.stop_ids |> Enum.concat(foxboro_extension.stop_ids) |> Enum.uniq()
+
+    [do_list_from_shapes(mainline.name, distinct_stop_ids, stops, route)]
+    |> merge_branch_list(0)
+  end
+
+  def list_from_shapes(shapes, stops, %Route{id: "CR-Fairmount"} = route, 1) do
+    mainline = Enum.at(shapes, 0)
```

Comment:
note to self: don't rely on the order, no reason to assume it won't change, instead look at the data